libdnf/transaction/db/archs.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "archs.hpp"
+
+namespace libdnf::transaction {
+
+static constexpr const char * SQL_NAME_INSERT_IF_NOT_EXISTS = R"**(
+    INSERT INTO
+        archs (
+            name
+        )
+    VALUES
+        (?)
+    ON CONFLICT DO NOTHING
+    RETURNING id
+)**";
+
+std::unique_ptr<utils::SQLite3::Statement> archs_insert_if_not_exists_new_query(utils::SQLite3 & conn) {
+    return std::make_unique<utils::SQLite3::Statement>(conn, SQL_NAME_INSERT_IF_NOT_EXISTS);
+}
+
+int64_t archs_insert_if_not_exists(utils::SQLite3::Statement & query, const std::string & name) {
+    int64_t result = 0;
+
+    query.bindv(name);
+    if (query.step() == utils::SQLite3::Statement::StepResult::ROW) {
+        result = query.get<int64_t>(0);
+    }
+    query.reset();
+    return result;
+}
+
+static constexpr const char * SQL_GET_ID = R"**(
+    SELECT
+        id
+    FROM
+        archs
+    WHERE
+        name = ?
+)**";
+
+std::unique_ptr<utils::SQLite3::Statement> archs_get_id_new_query(utils::SQLite3 & conn) {
```

Comment:
Unused, probably to be deleted? Or reference in `hpp`?

Reply:
It's a remnant. I was thinking about the optimal solution for working with tables. I'll delete, internal optimizations can be solved later.